src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -2373,5 +2366,334 @@ class Attr : System.Attribute { public Attr(string s) {} }";
+        }
+
+        [Fact]
+        public void OpenTypeInNameof_CSharp13_Nested1()
+        {
+            CreateCompilation("""
+                using System;
+                
+                var v = nameof(A<>.B<int>);
+                Console.WriteLine(v);
+
+                class A<X> { public class B<Y>; }
+                """, parseOptions: TestOptions.Regular13).VerifyDiagnostics(
+                // (3,16): error CS8652: The feature 'Unbound generic types in nameof operator' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                // var v = nameof(A<>.B<int>);
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "A<>").WithArguments("Unbound generic types in nameof operator").WithLocation(3, 16));
+        }
+
+        [Fact]
+        public void OpenTypeInNameof_CSharp13_Nested2()
+        {
+            CreateCompilation("""
+                using System;
+                
+                var v = nameof(A<int>.B<>);
+                Console.WriteLine(v);
+
+                class A<X> { public class B<Y>; }
+                """, parseOptions: TestOptions.Regular13).VerifyDiagnostics(
+                // (3,23): error CS8652: The feature 'Unbound generic types in nameof operator' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                // var v = nameof(A<int>.B<>);
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "B<>").WithArguments("Unbound generic types in nameof operator").WithLocation(3, 23));
+        }
+
+        [Fact]
+        public void OpenTypeInNameof_CSharp13_Nested3()
+        {
+            CreateCompilation("""
+                using System;
+                
+                var v = nameof(A<>.B<>);
+                Console.WriteLine(v);
+
+                class A<X> { public class B<Y>; }
+                """, parseOptions: TestOptions.Regular13).VerifyDiagnostics(
+                    // (3,16): error CS8652: The feature 'Unbound generic types in nameof operator' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    // var v = nameof(A<>.B<>);
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "A<>").WithArguments("Unbound generic types in nameof operator").WithLocation(3, 16),
+                    // (3,20): error CS8652: The feature 'Unbound generic types in nameof operator' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    // var v = nameof(A<>.B<>);
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "B<>").WithArguments("Unbound generic types in nameof operator").WithLocation(3, 20));
+        }
+
+        [Fact]
+        public void OpenTypeInNameof_BaseCase()
+        {
+            CompileAndVerify(
+                CreateCompilation("""
```

Comment:
nit: This and similar tests could be simplified by removing `CreateCompilation`: 
`CompileAndVerify("... source ..., expectedOutput: "...").VerifyDiagnostics();`

Reply:
Done.  Thanks!